src/client/input/sdl.c

```diff
@@ -1425,6 +1425,14 @@ IN_Controller_Init(qboolean notify_user)
 
 	if (!SDL_WasInit(SDL_INIT_GAMECONTROLLER | SDL_INIT_HAPTIC))
 	{
+
+#ifdef SDL_HINT_JOYSTICK_HIDAPI_PS4_RUMBLE	// extended input reports on PS controllers (enables gyro thru bluetooth)
```

Comment:
I wonder if it's worth some option the the Makefile for enabling these hints or if they're obscure enough that it is not necessary and we can expect the very few interested users to pass the option at the `make` command line or set the hint by specifying the corresponding environment variable when starting the game.

Reply:
Can I ask the reason?
If there's an "_#ifdef_" before them, it's because I wanted to retain compatibility with SDL 2.0.10, so YQ2 can continue to be compiled on a Raspberry Pi, for example.
It's an alternative to use _SDL_VERSION_ATLEAST()_.
It must be said that these hints, when supported and enabled, provide no conflict in the system and there's no practical advantage for having them disabled. If they are disabled by default, it's because they "break DirectInput handling for applications that don't use SDL", according to SDL documentation. But in practice, DirectInput is used only in very old games, and every person using a PS4/PS5 controller with their PC uses an input remapper to emulate XInput anyway (Steam input, DS4Windows, reWASD, etc), which probably uses the same HIDAPI, with the same side effect (solved by power cycling the controller).
Having "native input" provides YQ2 with an usability advantage, especially for gyro configuration, which is always a pain in input remappers. Again, if the user doesn't use any of this, having them disabled provides no advantage.

If you want, I can replace the "_ifdefs_" with "_SDL_VERSION_ATLEAST()_", for 2.0.22 in this case.

Reply:
> I wonder if it's worth some option the the Makefile for enabling these hints or if they're obscure enough that it is not necessary and we can expect the very few interested users to pass the option at the `make` command line or set the hint by specifying the corresponding environment variable when starting the game.

I'm personally interested in this as someone who not only enjoys quake, but also someone who plays it exclusively on controller. SDL2 is a great controller plugin, especially for offering gyro aiming which is feel benefits quake on controller exponentially as seen in the quake remaster on ps4/5 and switch.

Reply:
I totally missunderstood this. I didn't realize that the HINT_* defines are defined by the SDL headers. :/